src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -293,87 +311,95 @@ ELSE NULL
 LEFT OUTER JOIN pg_type AS basetyp ON (basetyp.oid = typ.typbasetype)
 LEFT OUTER JOIN pg_description AS des ON des.objoid = cls.oid AND des.objsubid = attnum
 WHERE
-    relkind = 'r' AND
-    nspname NOT IN ('pg_catalog', 'information_schema') AND
-    attnum > 0 {tableFilter} ORDER BY attnum";
-
-                command.CommandText = commandText;
+  relkind = 'r' AND
+  nspname NOT IN ('pg_catalog', 'information_schema') AND
+  attnum > 0 AND
+  cls.relname <> '{HistoryRepository.DefaultTableName}'
+  {tableFilter}
+ORDER BY attnum";
+
+            using (var command = new NpgsqlCommand(commandText, connection))
+            using (var reader = command.ExecuteReader())
+            {
+                var tableGroups =
+                    reader.Cast<DbDataRecord>()
+                          .GroupBy(ddr =>
+                          (
+                              tableSchema: ddr.GetValueOrDefault<string>("nspname"),
+                              tableName: ddr.GetValueOrDefault<string>("relname")
+                          ));
 
-                using (var reader = command.ExecuteReader())
+                foreach (var tableGroup in tableGroups)
```

Comment:
Personally, I find non-trivial LINQ in the `foreach` header hard to read...

I'm going to align as below, but I'll circle back in another PR to take a pass at method-izing the inner loop so that we can get a nice functional-looking call here.